ci/bats/bats.cue

```diff
@@ -0,0 +1,87 @@
+package bats
+
+import (
+	"dagger.io/dagger"
+
+	"universe.dagger.io/docker"
+	"universe.dagger.io/bash"
+)
+
+#Bats: {
+	// Source code
+	source: dagger.#FS
+
+	initScript: string | *null
+
+	// Environment variables to pass to bats
+	env: [string]: string
+
+	// Bats version
+	version: *"1.6.0" | string
+
+	// Mount points for the bats container
+	mounts: [name=string]: _
+
+	docker.#Build & {
+		_packages: ["yarn", "git", "docker", "curl"]
+		_batsMods: ["bats-support", "bats-assert"]
+
+		steps: [
+			docker.#Pull & {
+				source: "bats/bats:\(version)"
+			},
+
+			// Symlink bash so we can `bash.#Run` entrypoint can work
+			docker.#Run & {
+				entrypoint: []
+				command: {
+					name: "ln"
+					args: ["-sf", "/usr/local/bin/bash", "/bin/bash"]
+				}
+			},
+
+			docker.#Copy & {
+				contents: source
+				dest:     "/src"
+			},
+
+			for pkg in _packages {
+				docker.#Run & {
+					entrypoint: []
+					command: {
+						name: "apk"
+						args: ["add", "--no-cache", pkg]
+					}
+				}
+			},
+
+			for mod in _batsMods {
+				docker.#Run & {
+					entrypoint: []
+					workdir: "/src"
+					command: {
+						name: "yarn"
+						args: ["add", mod]
```

Comment:
This will remove the version pinning from `package.json` -- why not just `yarn` from the directory instead?

Reply:
as discussed out loud, I'll keep this one as-is (context: the intent is to remove the dependency on yarn for the user)